src/Avalonia.Build.Tasks/XamlCompilerTaskExecutor.cs

```diff
@@ -114,6 +123,16 @@ MessageImportance logImportance
                 // Nothing to do
                 return null;
 
+            if (asm.MainModule.TryGetTypeReference("System.Reflection.AssemblyMetadataAttribute", out var asmMetadata))
```

Comment:
TryGetTypeReference seems to only look for types that are already imported to the target module. I think should reuse type resolving logic from CecilTypeSystem.